rust/kernel/sync/guard.rs

```diff
@@ -144,26 +144,16 @@ pub trait LockFactory {
 /// A lock that can be initialised with a single lock class key.
 pub trait LockIniter {
     /// Initialises the lock instance so that it can be safely used.
-    ///
-    /// # Safety
-    ///
-    /// `key` must point to a valid memory location that will remain valid until the lock is
-    /// dropped.
-    unsafe fn init_lock(
-        self: Pin<&mut Self>,
-        name: &'static CStr,
-        key: *mut bindings::lock_class_key,
-    );
+    fn init_lock(self: Pin<&mut Self>, name: &'static CStr, key: &'static LockClassKey);
```

Comment:
Is this safe to be called on an already initialized lock?

Reply:
IIRC, you asked the same question when this was first introduced :)

Since we require a mutable reference, no one else should be using the lock, so calling this again is ok.

Reply:
Forgot about that 😅